.github/workflows/backwards_compatibility_tests_workflow.yml

```diff
@@ -31,12 +32,43 @@ jobs:
         with:
           java-version: ${{ matrix.java }}
 
-      - name: Install dependencies
+      - name: Install dependencies on ubuntu
+        if: startsWith(matrix.os,'ubuntu')
         run: |
           sudo apt-get install libopenblas-dev gfortran -y
 
+      - name: Install MinGW Using Scoop
+        if: startsWith(matrix.os,'windows')
+        run: |
+          iex "& {$(irm get.scoop.sh)} -RunAsAdmin"
+          scoop bucket add main
+          scoop install mingw
+
+      - name: Add MinGW to PATH
+        if: startsWith(matrix.os,'windows')
+        run: |
+          echo "C:/Users/runneradmin/scoop/apps/mingw/current/bin" >> $env:GITHUB_PATH
+          refreshenv
+
+      - name: Download OpenBLAS
+        if: startsWith(matrix.os,'windows')
+        run: |
+          curl -L -O https://github.com/xianyi/OpenBLAS/releases/download/v0.3.21/OpenBLAS-0.3.21-x64.zip
+          mkdir OpenBLAS
+          Expand-Archive -Path .\OpenBLAS-0.3.21-x64.zip -DestinationPath .\OpenBLAS\
+          mkdir ./src/main/resources/windowsDependencies
+          cp ./OpenBLAS/bin/libopenblas.dll ./src/main/resources/windowsDependencies/
+          rm .\OpenBLAS-0.3.21-x64.zip
+          rm -r .\OpenBLAS\
 
-      - name: Run k-NN Restart-Upgrade BWC Tests from BWCVersion-${{ matrix.bwc_version }} to OpenSearch Version-${{ matrix.opensearch_version }}
+      - name: Run k-NN Restart-Upgrade BWC Tests from BWCVersion-${{ matrix.bwc_version }} to OpenSearch Version-${{ matrix.opensearch_version }} Windows
+        if: (startsWith(matrix.os,'windows') && (startsWith(matrix.bwc_version, '1.3') || startsWith(matrix.bwc_version, '2.4') || startsWith(matrix.bwc_version, '2.5') || startsWith(matrix.bwc_version, '2.6') || startsWith(matrix.bwc_version, '3.')))
```

Comment:
Why cant we use matrix here? This will become a burden to support in future. Alternatively, can we switch to say which versions are not supported by Windows?

Reply:
I found an option to exclude the versions from given matrix for windows